File: WaterManager/src/main/java/pissir/watermanager/model/item/Attivazione.java
package pissir.watermanager.model.item;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


/**
 * @author Alessandro Gattico
 */
@Setter
@Getter
@NoArgsConstructor
public class Attivazione {
	
	private int id;
	private boolean current;
	private String time;
	private int idAttuatore;
	
	
	public Attivazione(boolean current, String time, int idAttuatore) {
		this.current = current;
		this.time = time;
		this.idAttuatore = idAttuatore;
	}
	
	
	public Attivazione(int id, String data, int stato, int idAttuatore) {
		this.id = id;
		this.time = data;
		if (stato > 0) {
			this.current = true;
		} else {
			this.current = false;
		}
		this.idAttuatore = idAttuatore;
	}
	
}

File: WaterManagerUI/Model/Item/Attivazione.cs
using Newtonsoft.Json;

namespace WaterManagerUI.Model.Item;

public class Attivazione
{
    public int id { get; set; }
    public bool current { get; set; }
    public String time { get; set; }
    public int idAttuatore { get; set; }

    public Attivazione()
    {
    }

    public Attivazione(String data, bool stato, int idAttuatore)
    {
        this.id = 0;
        this.time = data;
        this.current = stato;
        this.idAttuatore = idAttuatore;
    }

    [JsonConstructor]
    public Attivazione(int id, bool current, string time, int idAttuatore)
    {
        this.id = id;
        this.current = current;
        this.time = time;
        this.idAttuatore = idAttuatore;
    }

    public DateTime attivazioneAsDateTime()
    {
        DateTime attivazioneDate;
        if (DateTime.TryParse(time, out attivazioneDate))
        {
            return attivazioneDate;
        }
        else
        {
            return DateTime.MinValue;
        }
    }
}